source/core/Array.cpp

```diff
@@ -1569,15 +1561,15 @@ VIREO_FUNCTION_SIGNATURE4(ArrayThreshold, Double, TypedArrayCoreRef, Double, Int
         }
     } else {
         for (IntIndex i = startIndex; i < arrayIn->Length(); i++) {
-            Double y = ReadDoubleFromMemory(arrayIn->ElementType(), arrayIn->BeginAt(i));
-            if (i == startIndex) { left = right = (Double)startIndex;}
+            const Double y = ReadDoubleFromMemory(arrayIn->ElementType(), arrayIn->BeginAt(i));
+            if (i == startIndex) { right = static_cast<Double>(startIndex);}
             if (y < thresholdY) {
```

Comment:
I tend to think initialing left and right in one line is cleaner/clearer to read. Is there a clang-tidy rule that enforces this?

Reply:
You can see from his PR comment that Resharper is being used locally, not clang-tidy. We do not have clang-tidy configured locally or on the CI.

Reply:
I am not sure what his configuration is. If clangtools is installed, Resharper can be configured to check for clang-tidy issues.  I would like for us to enable first only rules that we can enforce with automated build. This particular one found by Resharper does not seem to be a quite safe one to apply. If left is used below before assignment in code maintenance, it is not a warning of any sort. Here, the programmer's intent of initializing both left and right seems more maintainable. So, if it is just a Resharper suggestion, we should not simply apply it. After all, it is not enforceable anyway in automated builds.

Reply:
As mentioned in my comment in https://github.com/ni/VireoSDK/pull/572, most of his changes actually seem to correspond to clang-tidy rules.

Reply:
Yeah, resharper uses clang-tidy to supply some of its rules internally. And when you install resharper it comes with the clangtools I believe, because I didn't do any additional setup to get those clan-tidy rules.

And you're right, some of these were just suggestions while some were warnings.

Reply:
As to the original concern over whether it is cleaner, the specific resharper hint was because the variable was always being reassigned and so initializing it here didn't matter.